src/summary/history/allergy-card.component.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { match } from "react-router";
+import SummaryCard from "../cards/summary-card.component";
+import SummaryCardRow from "../cards/summary-card-row.component";
+import SummaryCardRowContent from "../cards/summary-card-row-content.component";
+import { openmrsFetch } from "@openmrs/esm-api";
+import style from "./allergy-card-style.css";
+
+export default function AllergyCard(props: AllergyCardProps) {
+  const [patientAllergy, setPatientAllergy] = React.useState(null);
+
+  React.useEffect(() => {
+    getPatientAllergy(props.currentPatient.identifier[0].value)
+      .then(allergy => setPatientAllergy(allergy.data))
+      .catch(error => {
+        throw error;
+      });
+  }, [props.match]);
```

Comment:
unsubscribe from this observable once the component is unmounted, you can achieve that by 
```
const sub = getPatientAllergy()...
...
return ()=> sub.unsubscribe();
```